ide/editor.lib2/src/org/netbeans/modules/editor/lib2/view/DocumentViewOp.java

```diff
@@ -1227,10 +1227,12 @@ float getAvailableWidth() {
             setStatusBits(AVAILABLE_WIDTH_VALID);
             availableWidth = Integer.MAX_VALUE;
             renderWrapWidth = availableWidth;
-            TextLayout lineContTextLayout = getLineContinuationCharTextLayout();
-            if (lineContTextLayout != null && (getLineWrapType() != LineWrapType.NONE)) {
-                availableWidth = Math.max(getVisibleRect().width, 4 * getDefaultCharWidth() + lineContTextLayout.getAdvance());
-                renderWrapWidth = availableWidth - lineContTextLayout.getAdvance();
+            if (getLineWrapType() != LineWrapType.NONE) {
+                final TextLayout lineContTextLayout = getLineContinuationCharTextLayout();
+                final float lineContTextLayoutAdvance =
+                    lineContTextLayout == null ? 0f : lineContTextLayout.getAdvance();
+                availableWidth = Math.max(getVisibleRect().width, 4 * getDefaultCharWidth() + lineContTextLayoutAdvance);
```

Comment:
original code did not change `availableWidth` if `lineContTextLayout == null`; is it OK to alter it ?

Reply:
Yes; the getLineWrapType() != LineWrapType.NONE check is sufficient. Previously, getLineContinuationCharTextLayout would never return null in practice--only if the font could not display the line continuation character or its alternate. And if that ever happened, there would have been an NPE when WrapInfo calls paintTextLayout with the null layout.

I changed this so that getLineContinuationCharTextLayout can now return null without error (and added Javadoc to that effect). Rather than disabling line wrapping altogether, this simply avoids displaying displaying the line continuation character. Then I made this the default, unless the user opts to display hidden characters (paragraph marks etc).

I just looked over the call sites of getLineContinuationCharTextLayout  again; they seem to do the right thing.